### ISSUE TITLE
isLocalPlayer not set to true for local players

### ISSUE THREAD

**Issue by Plenopterix** [rempelj/unet-steamworks#6]
On objects that extend NetworkBehaviour there is a variable isLocalPlayer that is normally true for the local player only. This is used, for example, to prevent you from controlling other players on your screen when you try to move.

Perhaps somewhere in the SteamNetworkManager class it must be set to true but I don't know how to fix this. I use isLocalPlayer extensively.

**Comment by jespertheend** [comment]
OnStartLocalPlayer() is also not being called. I think there's more needed than just setting isLocalPlayer to true.

**Comment by jespertheend** [comment]
I fixed it by replacing `NetworkServer.SpawnWithClientAuthority(player, conn);` with `NetworkServer.AddPlayerForConnection(conn, player, 0);`

https://github.com/rempelj/unet-steamworks/blob/20836035e61c837b83b48ed7ad026105060f74dc/Assets/Scripts/UNETSteamworks/Network/UNETServerController.cs#L132

**Comment by Plenopterix** [comment]
Hey Jesper. Thanks, that helped. Have you tested this with hosts and clients or a server and clients? When I try to connect as a client it only starts another server.

**Comment by jespertheend** [comment]
I tried it with host and clients. Not sure how this would work as a server. Though I’m not really sure why you would want to run steam p2p as a server. I thought the idea of steamworks p2p was that you can connect multiple players without needing dedicated servers.

**Comment by Plenopterix** [comment]
Yes, you're right. It is p2p so it would not make sense to have a server and clients. I forgot about that.

It would be very helpful to know how you tested it. When I test it with two instances on my local machine, the second player starts another server instead of joining the client. How did you test? Did you use two separate computers with their own steam accounts?

**Comment by jespertheend** [comment]
Yes, I have a macbook and a PC so I connect the two to each other with two
different steam accounts.

**Comment by Plenopterix** [comment]
Great. Now I know why it hasn't been working for me. Thanks for your help.
